### ISSUE TITLE
nosetests: error: Error reading config file 'setup.cfg': no such option 'exclude-dir'

### ISSUE THREAD

**Issue by jGaboardi** [pysal/pysal#1072]
pysal/pysal#521 seems to have cropped up again in [`pysal v2.0.0rc3`](https://github.com/pysal/pysal/releases/tag/2.0.0rc3).

```
(pysal2.0rc3) user: $ nosetests -v pysal
Usage: nosetests [options]

nosetests: error: Error reading config file 'setup.cfg': no such option 'exclude-dir'
```

Is this an actual issue or just simply a local fix as per pysal/pysal#521?

**Comment by ljwolf** [comment]
We don't need `nose-exclude` anymore since we don't have `pysal.contrib`. I think we should remove that dependency.

**Comment by jGaboardi** [comment]
just realized I was this was a mistake on my part anyway. I was testing from the wrong directory.

**Comment by jGaboardi** [comment]
However, I am still having some trouble actually running `nosetests`. For all of the following:

```
(pysal2.0rc3)User:pysal-2.0.0rc3 $ nosetests -v pysal.explore.spaghetti
```
```
(pysal2.0rc3)User:pysal-2.0.0rc3 $ nosetests -v pysal.explore
```
```
(pysal2.0rc3)User:pysal-2.0.0rc3 $ nosetests -v pysal
```

0 tests are being run:
```
----------------------------------------------------------------------
Ran 0 tests in 0.002s

OK
```

Am I missing something here or should we not be able to run `nosetests` locally?

**Comment by ljwolf** [comment]
can you try with `pytest ./pysal`?

**Comment by jGaboardi** [comment]
@ljwolf @sjsrey that worked, but...

The following error crops up 128 times times:
```
E   ImportError: Python is not installed as a framework. 
     The Mac OS X backend will not be able to function
     correctly if Python is not installed as a framework.
     See the Python documentation for more information
     on installing Python as a framework on Mac OS X.
     Please either reinstall Python as a framework, or
     try one of the other backends. If you are using
     (Ana)Conda please install python.app and replace 
     the use of 'python' with 'pythonw'. See 'Working
     with Matplotlib on OSX' in the Matplotlib FAQ for
     more information.
```
with the guilty party apparently being:
```python
from matplotlib import pyplot as plt
```

full trace back (of one of the errors)

```
(pysal2.0rc3) user:pysal-2.0.0rc3 user $ pytest ./pysal
==================================================================== test session starts =====================================================================
platform darwin -- Python 3.6.7, pytest-4.1.1, py-1.7.0, pluggy-0.8.1
rootdir: /Users/USER/Desktop/pysal-2.0.0rc3, inifile:
collected 0 items / 128 errors                                                                                                                               

=========================================================================== ERRORS ===========================================================================
__________________________________________________ ERROR collecting pysal/explore/esda/tests/test_gamma.py ___________________________________________________
ImportError while importing test module '/Users/USER/Desktop/pysal-2.0.0rc3/pysal/explore/esda/tests/test_gamma.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
pysal/__init__.py:3: in <module>
    from . import explore
pysal/explore/__init__.py:4: in <module>
    from . import pointpats
pysal/explore/pointpats/__init__.py:4: in <module>
    from .pointpattern import PointPattern
pysal/explore/pointpats/pointpattern.py:12: in <module>
    from matplotlib import pyplot as plt
../../miniconda3/envs/pysal2.0rc3/lib/python3.6/site-packages/matplotlib/pyplot.py:2374: in <module>
    switch_backend(rcParams["backend"])
../../miniconda3/envs/pysal2.0rc3/lib/python3.6/site-packages/matplotlib/pyplot.py:207: in switch_backend
    backend_mod = importlib.import_module(backend_name)
../../miniconda3/envs/pysal2.0rc3/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
../../miniconda3/envs/pysal2.0rc3/lib/python3.6/site-packages/matplotlib/backends/backend_macosx.py:14: in <module>
    from matplotlib.backends import _macosx
E   ImportError: Python is not installed as a framework. The Mac OS X backend will not be able to function correctly if Python is not installed as a framework. See the Python documentation for more information on installing Python as a framework on Mac OS X. Please either reinstall Python as a framework, or try one of the other backends. If you are using (Ana)Conda please install python.app and replace the use of 'python' with 'pythonw'. See 'Working with Matplotlib on OSX' in the Matplotlib FAQ for more information.
```

**Comment by jGaboardi** [comment]
Also, there are 352 warnings for the following:
```
/Users/user/Desktop/pysal-2.0.0rc3/pysal/lib/io/util/shapefile.py:636: DeprecationWarning: fromstring() is deprecated. Use frombytes() instead.
    _unpackDict2(record, contentStruct, dat)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 128 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================================== 352 warnings, 128 error in 16.53 seconds ==========================================================

```

**Comment by jGaboardi** [comment]
pysal/pysal#1073

**Comment by jGaboardi** [comment]
@ljwolf I got `nosetests` running (was my mistake on downloading/checking out a branch). However, `nosetests` fail with a `c++` `Abort trap`. 

```
test_rose (pysal.explore.giddy.tests.test_directional.Rose_Tester) ... 2019-01-21 16:48:21.528 python[45679:2184502] -[NSApplication _setup:]: unrecognized selector sent to instance 0x7f915cf6db90
2019-01-21 16:48:21.540 python[45679:2184502] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSApplication _setup:]: unrecognized selector sent to instance 0x7f915cf6db90'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff3b89becd __exceptionPreprocess + 256
	1   libobjc.A.dylib                     0x00007fff67957720 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff3b919275 -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x00007fff3b83db40 ___forwarding___ + 1486
	4   CoreFoundation                      0x00007fff3b83d4e8 _CF_forwarding_prep_0 + 120
	5   libtk8.6.dylib                      0x0000001a2fa44475 TkpInit + 310
	6   libtk8.6.dylib                      0x0000001a2f9b4754 Initialize + 2457
	7   _tkinter.cpython-36m-darwin.so      0x0000001a2f816a46 _tkinter_create + 1174
	8   python                              0x000000010ebf1678 _PyCFunction_FastCallDict + 200
	9   python                              0x000000010ecc85ef call_function + 143
	10  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	11  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	12  python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	13  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	14  python                              0x000000010eb975b8 method_call + 136
	15  python                              0x000000010eb774fe PyObject_Call + 62
	16  python                              0x000000010ec18995 slot_tp_init + 117
	17  python                              0x000000010ec1cee1 type_call + 241
	18  python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	19  python                              0x000000010eb78067 _PyObject_FastCallKeywords + 327
	20  python                              0x000000010ecc86e8 call_function + 392
	21  python                              0x000000010ecc620c _PyEval_EvalFrameDefault + 46972
	22  python                              0x000000010ecc89ac fast_function + 188
	23  python                              0x000000010ecc864c call_function + 236
	24  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	25  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	26  python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	27  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	28  python                              0x000000010eb975b8 method_call + 136
	29  python                              0x000000010eb774fe PyObject_Call + 62
	30  python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	31  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	32  python                              0x000000010ecc8a5a fast_function + 362
	33  python                              0x000000010ecc864c call_function + 236
	34  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	35  python                              0x000000010ecc89ac fast_function + 188
	36  python                              0x000000010ecc864c call_function + 236
	37  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	38  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	39  python                              0x000000010ecc8a5a fast_function + 362
	40  python                              0x000000010ecc864c call_function + 236
	41  python                              0x000000010ecc620c _PyEval_EvalFrameDefault + 46972
	42  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	43  python                              0x000000010ecc8a5a fast_function + 362
	44  python                              0x000000010ecc864c call_function + 236
	45  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	46  python                              0x000000010ecc89ac fast_function + 188
	47  python                              0x000000010ecc864c call_function + 236
	48  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	49  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	50  python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	51  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	52  python                              0x000000010eb975b8 method_call + 136
	53  python                              0x000000010eb774fe PyObject_Call + 62
	54  python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	55  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	56  python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	57  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	58  python                              0x000000010eb975b8 method_call + 136
	59  python                              0x000000010eb774fe PyObject_Call + 62
	60  python                              0x000000010ec173b1 slot_tp_call + 113
	61  python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	62  python                              0x000000010ecc86e8 call_function + 392
	63  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	64  python                              0x000000010ecc89ac fast_function + 188
	65  python                              0x000000010ecc864c call_function + 236
	66  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	67  python                              0x000000010ecc90ff _PyFunction_FastCallDict + 175
	68  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	69  python                              0x000000010eb975b8 method_call + 136
	70  python                              0x000000010eb774fe PyObject_Call + 62
	71  python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	72  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	73  python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	74  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	75  python                              0x000000010eb975b8 method_call + 136
	76  python                              0x000000010eb774fe PyObject_Call + 62
	77  python                              0x000000010ec173b1 slot_tp_call + 113
	78  python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	79  python                              0x000000010ecc86e8 call_function + 392
	80  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	81  python                              0x000000010ecc90ff _PyFunction_FastCallDict + 175
	82  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	83  python                              0x000000010eb975b8 method_call + 136
	84  python                              0x000000010eb774fe PyObject_Call + 62
	85  python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	86  python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	87  python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	88  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	89  python                              0x000000010eb975b8 method_call + 136
	90  python                              0x000000010eb774fe PyObject_Call + 62
	91  python                              0x000000010ec173b1 slot_tp_call + 113
	92  python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	93  python                              0x000000010ecc86e8 call_function + 392
	94  python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	95  python                              0x000000010ecc90ff _PyFunction_FastCallDict + 175
	96  python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	97  python                              0x000000010eb975b8 method_call + 136
	98  python                              0x000000010eb774fe PyObject_Call + 62
	99  python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	100 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	101 python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	102 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	103 python                              0x000000010eb975b8 method_call + 136
	104 python                              0x000000010eb774fe PyObject_Call + 62
	105 python                              0x000000010ec173b1 slot_tp_call + 113
	106 python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	107 python                              0x000000010ecc86e8 call_function + 392
	108 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	109 python                              0x000000010ecc90ff _PyFunction_FastCallDict + 175
	110 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	111 python                              0x000000010eb975b8 method_call + 136
	112 python                              0x000000010eb774fe PyObject_Call + 62
	113 python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	114 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	115 python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	116 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	117 python                              0x000000010eb975b8 method_call + 136
	118 python                              0x000000010eb774fe PyObject_Call + 62
	119 python                              0x000000010ec173b1 slot_tp_call + 113
	120 python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	121 python                              0x000000010ecc86e8 call_function + 392
	122 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	123 python                              0x000000010ecc90ff _PyFunction_FastCallDict + 175
	124 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	125 python                              0x000000010eb975b8 method_call + 136
	126 python                              0x000000010eb774fe PyObject_Call + 62
	127 python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	128 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	129 python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	130 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	131 python                              0x000000010eb975b8 method_call + 136
	132 python                              0x000000010eb774fe PyObject_Call + 62
	133 python                              0x000000010ec173b1 slot_tp_call + 113
	134 python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	135 python                              0x000000010ecc86e8 call_function + 392
	136 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	137 python                              0x000000010ecc90ff _PyFunction_FastCallDict + 175
	138 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	139 python                              0x000000010eb975b8 method_call + 136
	140 python                              0x000000010eb774fe PyObject_Call + 62
	141 python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	142 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	143 python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	144 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	145 python                              0x000000010eb975b8 method_call + 136
	146 python                              0x000000010eb774fe PyObject_Call + 62
	147 python                              0x000000010ec173b1 slot_tp_call + 113
	148 python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	149 python                              0x000000010ecc86e8 call_function + 392
	150 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	151 python                              0x000000010ecc90ff _PyFunction_FastCallDict + 175
	152 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	153 python                              0x000000010eb975b8 method_call + 136
	154 python                              0x000000010eb774fe PyObject_Call + 62
	155 python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	156 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	157 python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	158 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	159 python                              0x000000010eb975b8 method_call + 136
	160 python                              0x000000010eb774fe PyObject_Call + 62
	161 python                              0x000000010ec173b1 slot_tp_call + 113
	162 python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	163 python                              0x000000010ecc86e8 call_function + 392
	164 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	165 python                              0x000000010ecc90ff _PyFunction_FastCallDict + 175
	166 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	167 python                              0x000000010eb975b8 method_call + 136
	168 python                              0x000000010eb774fe PyObject_Call + 62
	169 python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	170 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	171 python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	172 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	173 python                              0x000000010eb975b8 method_call + 136
	174 python                              0x000000010eb774fe PyObject_Call + 62
	175 python                              0x000000010ec173b1 slot_tp_call + 113
	176 python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	177 python                              0x000000010ecc86e8 call_function + 392
	178 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	179 python                              0x000000010ecc89ac fast_function + 188
	180 python                              0x000000010ecc864c call_function + 236
	181 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	182 python                              0x000000010ecc89ac fast_function + 188
	183 python                              0x000000010ecc864c call_function + 236
	184 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	185 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	186 python                              0x000000010ebbab4d function_call + 125
	187 python                              0x000000010eb774fe PyObject_Call + 62
	188 python                              0x000000010ecc6352 _PyEval_EvalFrameDefault + 47298
	189 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	190 python                              0x000000010ecc91c5 _PyFunction_FastCallDict + 373
	191 python                              0x000000010eb6feb0 _PyObject_FastCallDict + 320
	192 python                              0x000000010eb975b8 method_call + 136
	193 python                              0x000000010eb774fe PyObject_Call + 62
	194 python                              0x000000010ec18995 slot_tp_init + 117
	195 python                              0x000000010ec1cee1 type_call + 241
	196 python                              0x000000010eb6fe21 _PyObject_FastCallDict + 177
	197 python                              0x000000010ecc86e8 call_function + 392
	198 python                              0x000000010ecc615c _PyEval_EvalFrameDefault + 46796
	199 python                              0x000000010ecb98c9 _PyEval_EvalCodeWithName + 425
	200 python                              0x000000010ed1210c PyRun_FileExFlags + 252
	201 python                              0x000000010ed115e4 PyRun_SimpleFileExFlags + 372
	202 python                              0x000000010ed38226 Py_Main + 3766
	203 python                              0x000000010eb67f29 main + 313
	204 libdyld.dylib                       0x00007fff68a25ed9 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6
```

**Comment by jGaboardi** [comment]
#1085

**Comment by sjsrey** [comment]
Closed with #1091.
